patches/tModLoader/Terraria/ModLoader/Default/ModAccessorySlotPlayer.cs

```diff
@@ -325,55 +317,55 @@ public override void OnEquipmentLoadoutSwitched(int oldLoadoutIndex, int loadout
 				}
 			}
-		}
 
-		DetectConflictsWithSharedSlots();
+			DetectConflictsWithSharedSlots();
+		}
 	}
 
-	private void DetectConflictsWithSharedSlots()
+	private void DetectConflictsWithSharedSlots() // Only called on local client.
 	{
-		// Might be easier to just sync exAccessorySlotLoadoutConflict than run this on other clients, especially if we need to support HasEquipmentLoadoutSupport being a client-side config. (either sync this or sync sharedLoadoutSlotTypes to support that)
-
 		// Handle conflicts that arise from supporting shared slots.
-		Item temp = new Item();
-		for (int i = 0; i < exAccessorySlot.Length; i++) {
+		bool anyConflict = false;
+		for (int i = 0; i < exAccessorySlot.Length / 2; i++) {
 			if (IsSharedSlot(i)) {
-				Utils.Swap(ref exAccessorySlot[i], ref temp); // Swap out item for check otherwise false negative.
-				if (!temp.IsAir && !Loader.ModSlotCheckForPlayer(Player, temp, i, Terraria.UI.ItemSlot.Context.ModdedAccessorySlot)) {
-					exAccessorySlotLoadoutConflict[i] = true;
-				}
-				else {
-					exAccessorySlotLoadoutConflict[i] = false;
-				}
-				Utils.Swap(ref exAccessorySlot[i], ref temp);
+				anyConflict |= Loader.IsAccessoryInConflict(Player, exAccessorySlot[i], i, Terraria.UI.ItemSlot.Context.ModdedAccessorySlot);
+				anyConflict |= Loader.IsAccessoryInConflict(Player, exAccessorySlot[i + SlotCount], i + SlotCount, Terraria.UI.ItemSlot.Context.ModdedVanityAccessorySlot);
 			}
 		}
 
-		if (Main.myPlayer == Player.whoAmI && exAccessorySlotLoadoutConflict.Any(x => x)) {
+		if (anyConflict) {
 			Main.NewText(Language.GetTextValue("tModLoader.SharedAccessorySlotConflictMessage"));
+			// TODO: Main.NewText doesn't work with ModPlayer.OnEnterWorld in multiplayer since the chat is cleared during the joining process. This will need to be fixed and also affects ExampleMod usages of OnEnterWorld as well.
 		}
 	}
 
 	// If we allow HasEquipmentLoadoutSupport to differ per-client, this might not be suitable without extra changes.
 	internal bool IsSharedSlot(int slotType)
 	{
-		return sharedLoadoutSlotTypes.Contains(slotType % slots.Count);
+		return sharedLoadoutSlotTypes.Contains(slotType % SlotCount);
 	}
 
 	internal bool SharedSlotHasLoadoutConflict(int slotType)
 	{
-		int functional = slotType % slots.Count;
-		int vanity = functional + slots.Count;
-		return exAccessorySlotLoadoutConflict[functional] || exAccessorySlotLoadoutConflict[vanity];
+		if (!IsSharedSlot(slotType))
+			return false;
+
+		int functional = slotType % SlotCount;
+		int vanity = functional + SlotCount;
+
+		// Handle conflicts that arise from supporting shared slots.
+		bool functionalIssue = Loader.IsAccessoryInConflict(Player, exAccessorySlot[functional], functional, Terraria.UI.ItemSlot.Context.ModdedAccessorySlot);
+		bool vanityIssue = Loader.IsAccessoryInConflict(Player, exAccessorySlot[vanity], vanity, Terraria.UI.ItemSlot.Context.ModdedVanityAccessorySlot);
+
+		return functionalIssue || vanityIssue;
```

Comment:
It would be good to disable the vanity and functional slots independently. I'm not sure how feasible that is.

Reply:
I was able to implement this appraoch.